src/frontend/apps/drive/src/pages/explorer/items/[id].tsx

```diff
@@ -1,30 +1,31 @@
-import { getDriver } from "@/features/config/Config";
 import { ItemFilters } from "@/features/drivers/Driver";
 import { AppExplorer } from "@/features/explorer/components/app-view/AppExplorer";
 import { getGlobalExplorerLayout } from "@/features/layouts/components/explorer/ExplorerLayout";
-import { useQuery } from "@tanstack/react-query";
+import { useInfiniteChildren } from "@/features/explorer/hooks/useInfiniteChildren";
 import { useRouter } from "next/router";
-import { useState } from "react";
+import { useState, useMemo } from "react";
 export default function ItemPage() {
   const router = useRouter();
   const itemId = router.query.id as string;
   const [filters, setFilters] = useState<ItemFilters>({});
 
-  const { data: itemChildren } = useQuery({
-    queryKey: [
-      "items",
-      itemId,
-      "children",
-      ...(Object.keys(filters).length ? [JSON.stringify(filters)] : []),
-    ],
-    queryFn: () => getDriver().getChildren(itemId, filters),
-  });
+  const { data, fetchNextPage, hasNextPage, isFetchingNextPage, isLoading } =
+    useInfiniteChildren(itemId, filters);
+
+  // Flatten all pages into a single array of items
+  const itemChildren = useMemo(() => {
+    return data?.pages.flatMap((page) => page.children) ?? [];
```

Comment:
Oh, weird that it is not flattened by default

Reply:
yep..